numpy/core/tests/test_multiarray.py

```diff
@@ -3377,6 +3378,18 @@ def fail(*args, **kwargs):
         y = np.fromfile(self.filename, dtype=self.dtype)
         assert_array_equal(y, self.x.flat)
 
+    def test_largish_file(self):
+        # check the fallocate path on files > 16MB
+        d = np.zeros(4 * 1024 ** 2)
```

Comment:
This won't actually trigger `fallocate`, since it is less than `16 * 1024 * 1024`. Is that intentional?